### PR TITLE
Add feature to generate `handle_reply` in bechmarks generator

### LINKED ISSUE #1244: Misuse of `gr_exit_code` & `gr_reply_to` in benchmarks
### Problem

The syscalls are executed in `handle` method.

### Steps

1. Reproduced in #1211 

### Possible Solution

_No response_

### Notes

_No response_

### Relevant Log Output

<details><summary>Click to expand/collapse</summary>
<p>

```
<Paste your log here>
```

</p>
</details>

### PR DESCRIPTION
**Pull request by gshep**
Resolves #1244 

- add optional reply_body field to ModuleDefinition
- fix gr_exit_code & gr_reply_to benchmarks by calling the function in `handle_reply`
- fix docs for `gr_exit_code`

@gear-tech/dev

### CODE REVIEW

File: pallets/gear/src/benchmarking/code.rs [hunk 1]
```diff
@@ -71,6 +71,9 @@ pub struct ModuleDefinition {
     /// Function body of the exported `handle` function. Body is empty if `None`.
     /// Its index is `imported_functions.len() + 1`.
     pub handle_body: Option<FuncBody>,
+    /// Function body of the exported `handle_reply` function. Body is missing if `None`.
+    /// Its index is `imported_functions.len() + 2`.
+    pub reply_body: Option<FuncBody>,
     /// Function body of a non-exported function with index `imported_functions.len() + 2`.
```
Comment (ukint-vs) [inline pallets/gear/src/benchmarking/code.rs:77]:
```suggestion
    /// Function body of a non-exported function with index `imported_functions.len() + 3`.
```

**Review by ukint-vs — APPROVED** [review]

File: pallets/gear/src/benchmarking/code.rs (hunk 1 above)
Comment (gshep) [reply pallets/gear/src/benchmarking/code.rs:77]: The index of the `handle` is `imported_functions.len() + 1`, so this I set to `handle`'s index + 1 = `imported_functions.len() + 2`. Am I wrong?

Comment (ukint-vs) [reply pallets/gear/src/benchmarking/code.rs:77]: I think better to make empty `handle_reply` if body is None.

Comment (gshep) [reply pallets/gear/src/benchmarking/code.rs:77]: ok

**Review by breathx — APPROVED** [review]

**Comment by breathx** [comment]
Related to failed CI: benchmark's "sandbox" not guaranteed to invoke only "handle" function

File: pallets/gear/src/benchmarking/code.rs (hunk 1 above)
Comment (breathx) [inline pallets/gear/src/benchmarking/code.rs:77]: it's also +2

Comment (ukint-vs) [reply pallets/gear/src/benchmarking/code.rs:77]: change this https://github.com/gear-tech/gear/blob/c8b553e22d2a8ced1d297a6b8a503847ae1505ac/pallets/gear/src/benchmarking/mod.rs#L1903

Comment (breathx) [reply pallets/gear/src/benchmarking/code.rs:77]: not only there
everywhere where "aux" appears: not only TableSegment, but Call(num)'s also

**Review by ukint-vs — APPROVED** [review]

**Pull request merged**